inc/Classes/Translation.php

```diff
@@ -300,79 +295,64 @@ public function xml_write_db_to_file($module)
 
         $content = '';
         $res = $db->qry('
-            SELECT * 
-            FROM %prefix%translation 
-            WHERE 
-              file = %string% 
-              AND obsolete = 0', $module);
+            SELECT
+                `id`,
+                `org`,
+                `de`,
+                `en`,
+                `es`,
+                `fr`,
+                `nl`,
+                `it`
+            FROM %prefix%translation
+            WHERE
+              `file` = %string%
+              AND `obsolete` = 0
+              AND `id` != \'\'', $module);
         while ($row = $db->fetch_array($res)) {
-            if ($row['id'] != '') {
-                $entry = $xml->write_tag('id', $row['id'], 4);
-                $entry .= $xml->write_tag('org', $row['org'], 4);
-
-                if ($row['de'] != '') {
-                    $entry .= $xml->write_tag('de', $row['de'], 4);
-                }
-
-                if ($row['en'] != '') {
-                    $entry .= $xml->write_tag('en', $row['en'], 4);
-                }
-
-                if ($row['es'] != '') {
-                    $entry .= $xml->write_tag('es', $row['es'], 4);
-                }
-
-                if ($row['fr'] != '') {
-                    $entry .= $xml->write_tag('fr', $row['fr'], 4);
-                }
-
-                if ($row['nl'] != '') {
-                    $entry .= $xml->write_tag('nl', $row['nl'], 4);
-                }
+            $entry = $xml->write_tag('id', $row['id'], 4);
+            $entry .= $xml->write_tag('org', $row['org'], 4);
 
-                if ($row['it'] != '') {
-                    $entry .= $xml->write_tag('it', $row['it'], 4);
+            $languagesToAdd = ['de', 'en', 'es', 'fr', 'nl', 'it'];
```

Comment:
Shouldn't this also be derived from the definition at top?

Reply:
Fixed in https://github.com/lansuite/lansuite/pull/654/commits/d3bd13d050946bd78ce428444e760f3b7d0121fc